modules/claims_api/spec/lib/claims_api/error/error_handler.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'claims_api/special_issue_mappers/bgs'
+require 'token_validation/v2/client'
+require 'claims_api/error/error_handler'
+# require 'spec_helper'
+
+# describe ApplicationController, type: :controller do
+#   let(:scopes) { %w[claim.write] }
+#   controller do
+#     def test_invalid_token
+#       raise ::Common::Exceptions::TokenValidationError
+#     end
+#     skip_before_action :authenticate
+#   end
+
+#   before :each do
+#     routes.draw do
+#       get 'test_invalid_token' => 'anonymous#test_invalid_token'
+#     end
+#   end
+
+#   it 'catches an invalid token' do
+#     with_okta_user(scopes) do |auth_header|
+#       p auth_header
+#       request.headers.merge!(auth_header) # https://github.com/rspec/rspec-rails/issues/1655
+#       p request.headers
+#       # byebug
+#       get :test_invalid_token
+#       puts response.body
+#       expect(response).to have_http_status(401)
+#       parsed_body = JSON.parse(response.body)
+#       expect(parsed_body['detail']).to eq('Invalid token.')
+#     end
+#   end
+# end
+
+describe ApplicationController, type: :request do
+  let(:veteran_id) { '1013062086V794840' }
+  let(:scopes) { %w[claim.write] }
+  let(:get_poa_path) { "/services/benefits/v2/veterans/#{veteran_id}/power-of-attorney" }
+
+  it 'catches an invalid token' do
+    with_okta_user(scopes) do |auth_header|
+      allow(JWT).to receive(:decode).and_raise(JWT::DecodeError)
+
+      get get_poa_path, headers: auth_header
+
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body['errors'][0]['detail']).to eq('Invalid token.')
+    end
+  end
+
+  it 'catches an expired token' do
```

Comment:
Dan and I tried that, but we could only get it to raise a `JWT::DecodeError` for some reason.

Reply:
The issue that we were running into was there are [three](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/openid_application_controller.rb#L326) [different](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/token.rb#L51) [times](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/token.rb#L30) `JWT.decode` is called before [the code we want it to hit](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/token.rb#L20) (i.e., where it catches and raises the `JWT::ExpiredSignature` error) and fail happens. It's a pretty brittle way of testing, so I'm totally in favor of a better approach that lets us still test a similar expectation. 

Austin had initially tried a different approach that used the `controller` keyword [within rspec](https://stackoverflow.com/questions/46986332/what-is-the-best-way-to-test-application-controller-rescue-status-codes-in-rails/47006072#47006072), but he was having different issues getting that to work. I went with the pattern I was more familiar with and then came up against the issue we solved for here. I'm all for trying to get the other technique working if it's not a time suck (seems much cleaner), but at the very least, adding some explanatory logic in a comment if this stays in.

Reply:
Maybe I'm missing something, but this code works for me locally.

```
  it 'catches an expired token' do
    with_okta_user(scopes) do |auth_header|
      allow(JWT).to receive(:decode).and_raise(JWT::ExpiredSignature)

      get get_poa_path, headers: auth_header

      parsed_body = JSON.parse(response.body)
      expect(parsed_body['errors'][0]['detail']).to eq('Invalid token.')
    end
  end
```

Reply:
Yes, that test should succeed with the error handling in place, but if you take out the error handling it still succeeds because the `ExpiredSignature` exception is getting converted to an `InvalidToken` somewhere. The test needs to throw an `ExpiredSignature` and needs to render the same as `InvalidToken`.